import "goby/common/protobuf/liaison_config.proto";

extend goby.common.protobuf.LiaisonConfig
{
  optional DRCGUIConfig drc_gui_config = 1100;
}

message DRCGUIConfig
{
  optional int32 latency_ms = 1 [default = 0];
  optional int32 drop_percentage = 2 [default = 0];

  optional bool do_rate_control = 10 [default = false];
  optional int32 max_rate = 11 [default = 0];

  enum RateUnits
  {
    BYTES_PER_SEC = 0;
    KBYTES_PER_SEC = 1;
    MBYTES_PER_SEC = 2;
    KBITS_PER_SEC = 3;
    MBITS_PER_SEC = 4;
  }
  
  optional RateUnits rate_unit = 12 [default = KBITS_PER_SEC];

  optional int32 tunnel_num = 100 [default = 51];

  enum TunnelType
  {
    LOOPBACK = 0;
    OPENVPN = 1;
  }
  optional TunnelType tunnel_type = 101 [default = LOOPBACK];

  optional string remote_address = 102;
  optional int32 port = 103;

  optional string tunnel_address = 104 [default = "172.19.5.1"];
  optional string remote_tunnel_address = 105 [default = "172.19.5.2"];
  
  optional string this_config_location = 200 [default = "/etc/drc_tc/drc_tc.pb.cfg"];
  
}

